### ISSUE TITLE
Publish a podspec.

### ISSUE THREAD

**Issue by agrosner** [tipsi/tipsi-stripe#137]
Instead of having us declare a separate target in our project for TPSStripe, please publish a podspec that we can consume directly without need for also specifying the stripe dependency.

**Comment by isnifer** [comment]
@agrosner we know we were wrong at 3.8.0. We are discussing how to do it better. Any ideas besides podspec?

**Comment by agrosner** [comment]
whats wrong with a podspec? most RN libs publish one and they work great.

**Comment by isnifer** [comment]
@agrosner npm is a common place for both platforms (iOS, Android)

**Comment by dmytrozavgorodniy** [comment]
@agrosner thanks for your advice! What RM libs do you mean?

**Comment by agrosner** [comment]
Yes but I was talking about including a podspec file in the repo. Not
publishing to cocoapods.

**Comment by anishtr4** [comment]
Even I am confused after stripe install via cocopod where should we copy the TPSStripe folder.

**Comment by isnifer** [comment]
@anishtr4 You don't need to copy anything. What are you talking about?

**Comment by anishtr4** [comment]
@isnifer  Iam using all my module via cococpods including stripe. I am little bit confused how to install this via cocopods. is there ".podspec" available , if no can you guide me to create one for this.

**Comment by anishtr4** [comment]
@isnifer  this is pod file i have created 
Pod::Spec.new do |s|
    s.name         = 'tipsi-stripe'
    s.author       = 'Tipsi'
    s.source_files = 'ios/TPSStripe/*'
    s.platform     = :ios, '8.0'
    s.license      = 'MIT'
    s.homepage     = 'https://github.com/tipsi/tipsi-stripe#readme'
    s.source       = {:path => './ios/*'}
    s.summary      = 'tipsi-stripe'
    s.version      = '3.8.1'
    s.dependency 'React'
end

but after installing the pods I am getting error saying #import <Stripe/Stripe.h> file not found

**Comment by agrosner** [comment]
yes. @anishtr4 that is why I filed this ticket. They should just publish a podspec and then we never have to import stripe pod directly, only this library.

**Comment by agrosner** [comment]
many other RN libraries do this:
https://github.com/wmcmahan/react-native-calendar-events
https://github.com/airbnb/lottie-react-native
https://github.com/react-native-community/react-native-blur
https://github.com/invertase/react-native-firebase
https://github.com/react-native-community/react-native-svg
The list goes on.

**Comment by anishtr4** [comment]
@agrosner   

I have fixed the issue . Use the following pod file.

Pod::Spec.new do |s|
    s.name         = 'tipsi-stripe'
    s.author       = 'Tipsi'
    s.source_files = 'ios/TPSStripe/*'
    s.platform     = :ios, '8.0'
    s.license      = 'MIT'
    s.homepage     = 'https://github.com/tipsi/tipsi-stripe#readme'
    s.source       = {:path => './ios/*'}
    s.summary      = 'tipsi-stripe'
    s.version      = '3.8.1'
    s.dependency 'Stripe'
    s.dependency 'React'
end

**Comment by anishtr4** [comment]
@agrosner  let me know if this works for you. :)

**Comment by isnifer** [comment]
@agrosner @anishtr4 I will talk with @dmytrozavgorodniy and he will add it tomorrow. Thank you. guys!

**Comment by dmytrozavgorodniy** [comment]
@agrosner @anishtr4 If we will use `.podspec` in dependency we should refer to [React cocoapods dependency](https://cocoapods.org/pods/React), but it is deprecated. React podspec refer to v0.13.0-rc of react native [(React.podspec.json)](https://github.com/CocoaPods/Specs/blob/master/Specs/5/0/c/React/0.13.0-rc/React.podspec.json)

**Comment by isnifer** [comment]
@agrosner @anishtr4 your turn)

**Comment by agrosner** [comment]
you dont reference the podspec for react on cocoapods....you reference the spec locally on your machine.

**Comment by agrosner** [comment]
this is my pod file in my project:
```
target 'my project' do

	pod 'RNFirebase', :path => '../node_modules/react-native-firebase'

	pod 'Firebase/Core'
	pod 'Firebase/Messaging'
	pod 'Firebase/Crash'

	pod "Yoga", :path => "../node_modules/react-native/ReactCommon/yoga"
	pod 'React', :path => '../node_modules/react-native', :subspecs => [
  'BatchedBridge', # Required For React Native 0.45.0+
  'Core',
	'DevSupport',
 	'RCTText',
	'RCTGeolocation',
	'RCTVibration',
	'RCTNetwork',
	'RCTWebSocket',
	'RCTImage',
	'RCTLinkingIOS',
	'RCTSettings',
	'RCTActionSheet',
	'RCTAnimation',
  # Add any other subspecs you want to use in your project
]

	pod 'Stripe', '~> 10.1.0'
	pod 'BVLinearGradient', :path => '../node_modules/react-native-linear-gradient'
	pod 'RNDeviceInfo', :path => '../node_modules/react-native-device-info'
	pod 'react-native-calendar-events', :path => '../node_modules/react-native-calendar-events'
	pod 'RNSVG', :path => '../node_modules/react-native-svg'
	pod 'react-native-blur', :path => '../node_modules/react-native-blur'
	pod 'lottie-ios', :path => '../node_modules/lottie-ios'
	pod 'lottie-react-native', :path => '../node_modules/lottie-react-native'
end
```

**Comment by shaksi** [comment]
having a .podspec is an easy win and one that gets rid of lot of confusion as it adds stripe to the podfile after link and one only has to run pod install from there.

thanks anish for the podspec.

**Comment by shaksi** [comment]
with the latest release this taken care of by the link script. (looks like it uses ruby script to do the necessary linking etc)

**Comment by agrosner** [comment]
react-native link does not work for me due to using Kotlin in the Android project. I manually add the libs as Pods in iOS also because it doesn't or didn't respect cocoapods, leading to broken iOS builds anyways. Adding a .podspec to the repo that you reference is a one-liner. Modifying the node_modules project in my app locally does not work on continuous integration environments.

**Comment by isnifer** [comment]
@agrosner latest release doesn't update project in node_modules

**Comment by isnifer** [comment]
@agrosner update to 3.9.0

**Comment by agrosner** [comment]
right but the whole point of the ticket is to provide a podspec, which seems to be missing in 3.9.0 and react-native link does not work for me (and potentially others who implement pods + Kotlin Android the way I do)

**Comment by isnifer** [comment]
@agrosner @shaksi @anishtr4 published as **4.1.0**
